src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -295,19 +298,40 @@ function createEnumTypeChecker(expectedValues) {
 
   function validate(props, propName, componentName, location, propFullName) {
     var propValue = props[propName];
+    var locationName = ReactPropTypeLocationNames[location];
     for (var i = 0; i < expectedValues.length; i++) {
       if (is(propValue, expectedValues[i])) {
         return null;
       }
+      if (
+        getPropType(expectedValues[i]) === 'function' &&
+        expectedValues[i].hasOwnProperty('expectedType')
+      ) {
+        return new PropTypeError(
+          `Invalid enum values for ` +
+          `${locationName} \`${propFullName}\` of value \`${propValue}\`. ` +
+          `Found React PropType: "${expectedValues[i].expectedType}". ` +
+          `Did you mean to use oneOfType instead?`
+        );
+      }
     }
 
-    var locationName = ReactPropTypeLocationNames[location];
-    var valuesString = JSON.stringify(expectedValues);
+    var valuesString = JSON.stringify(expectedValues, functionPrettifier);
     return new PropTypeError(
       `Invalid ${locationName} \`${propFullName}\` of value \`${propValue}\` ` +
       `supplied to \`${componentName}\`, expected one of ${valuesString}.`
     );
   }
+
+  function functionPrettifier(name, values) {
+    for (var i = 0; i < values.length; i++) {
+      if (getPropType(values[i]) === 'function') {
+        values[i] = 'Unexpected function: ' + values[i].toString();
+      }
+    }
+    return values;
```

Comment:
maybe something like this instead?
```js
return values.map(value => (
  getPropType(value) === 'function' ? `Unexpected function: ${value}` : value
));
```
You shouldn't have to specially call `toString` on `value`.